src/features/charts/redux/middleware.test.ts

```diff
@@ -0,0 +1,47 @@
+import { chartMiddleware } from './middleware'
+import { updateChartRange } from './actions'
+
+jest.mock('services/socketService');
+
+describe('Testing Chart middleware', () => {
+    let mockSocket: any;
+    let socketService: any;
+    let store: any;
+    let next: jest.Mock;
+
+    beforeEach(() => {
```

Comment:
beforeEach test mock our test dependencies ie. store, socket, next

mockSocket = the function get method, emit
socketService instantiates the get from the mock imported services module